File: AnkiDroid/src/main/java/com/ichi2/compat/CompatV30.java


package com.ichi2.compat;



import android.annotation.TargetApi;

import android.view.View;
import android.view.Window;
import android.view.WindowInsets;
import android.view.WindowInsetsController;

import com.ichi2.anki.R;
import com.ichi2.anki.reviewer.FullScreenMode;


/** Implementation of {@link Compat} for SDK level 30 and higher. Check {@link Compat}'s for more detail. */
@TargetApi(30)
public class CompatV30 extends CompatV26 implements Compat {
    @Override
    public void setFullscreen(Window window) {
        window.getDecorView().getWindowInsetsController().hide(WindowInsets.Type.statusBars());
    }

    @Override
    public void hideSystembars(Window window) {
        window.setDecorFitsSystemWindows(false);
        WindowInsetsController controller = window.getInsetsController();
        if (controller != null) {
            controller.hide(WindowInsets.Type.statusBars() | WindowInsets.Type.navigationBars());
            controller.setSystemBarsBehavior(WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
        }
    }

    @Override
    public boolean isImmersiveSystemUiVisible(Window window) {
        return (window.getDecorView().getWindowInsetsController().getSystemBarsAppearance()) == 0;
    }
}

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatV30.kt
package com.ichi2.compat

import android.annotation.TargetApi
import android.view.Window
import android.view.WindowInsets
import android.view.WindowInsetsController

/** Implementation of [Compat] for SDK level 30 and higher. Check [Compat]'s for more detail.  */
@TargetApi(30)
class CompatV30 : CompatV26(), Compat {
    override fun setFullscreen(window: Window?) {
        window?.decorView?.windowInsetsController!!.hide(WindowInsets.Type.statusBars())
    }

    override fun hideSystemBars(window: Window?) {
        window?.setDecorFitsSystemWindows(false)
        val controller = window?.insetsController
        if (controller != null) {
            controller.hide(WindowInsets.Type.statusBars() or WindowInsets.Type.navigationBars())
            controller.systemBarsBehavior =
                WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        }
    }

    override fun isImmersiveSystemUiVisible(window: Window?): Boolean {
        return window?.decorView?.windowInsetsController!!.systemBarsAppearance == 0
    }
}
